### PR TITLE
POD-2310: Remove SA from Workspace

### PR DESCRIPTION
**Pull request by sahakiann**
[POD-2310](https://broadinstitute.atlassian.net/browse/POD-2310)

Example workspace: https://app.terra.bio/#workspaces/ops-integration-billing/ns_testing_2_Staging

Example logging showing removal of old SA: 
```
INFO: 2024-11-26 10:49:36,376 : Removing access for the service account associated with the OLD dataset from workspace: 'tdr-ingest-sa@datarepo-62ea4fa9.iam.gserviceaccount.com'
INFO: 2024-11-26 10:49:36,376 : Updating user tdr-ingest-sa@datarepo-62ea4fa9.iam.gserviceaccount.com to NO ACCESS in workspace ops-integration-billing/ns_testing_2_Staging
```

[POD-2310]: https://broadinstitute.atlassian.net/browse/POD-2310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by snovod — CHANGES_REQUESTED** [review]
I think we only to remove the OLD data_ingest_sa when we are deleting older dataset

**Review by John-Scira — APPROVED** [review]

File: python/set_up_staging_workspace_and_dataset.py [hunk 1]
```diff
@@ -519,7 +525,19 @@ def run(self) -> list[WorkflowConfigs]:
         auth_group=auth_group,
         controlled_access=controlled_access,
         delete_existing_dataset=delete_existing_dataset
-    ).run()
+    )
+    if delete_existing_dataset:
```
[the remark this answers is not in the document]
Comment (sahakiann) [reply python/set_up_staging_workspace_and_dataset.py:529]: Hm not sure what you mean - the `get_sa_for_dataset_to_delete()` method is being called before the `run()` method. So it removes access for the SA associated with the dataset that's about to get deleted before creating the new dataset.

File: python/set_up_staging_workspace_and_dataset.py [hunk 2]
```diff
@@ -519,7 +525,19 @@ def run(self) -> list[WorkflowConfigs]:
...
+        dataset_to_delete_metadata = dataset_setup.get_sa_for_dataset_to_delete()
+        if dataset_to_delete_metadata:
+            data_project = dataset_to_delete_metadata[0]["dataProject"]
```
[the remark this answers is not in the document]
Comment (sahakiann) [reply python/set_up_staging_workspace_and_dataset.py:532]: Good point, updated! Forgot that was a method that existed

File: python/set_up_staging_workspace_and_dataset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (snovod) [reply python/set_up_staging_workspace_and_dataset.py:532]: Sorry, I completyly missed that it is initiliatzed but run isn't called.

**Review by snovod — APPROVED** [review]

**Pull request merged**